erpnext/accounts/report/accounts_receivable/accounts_receivable.py

```diff
@@ -578,6 +578,26 @@ def get_gl_entries(self):
 				{1} {2}"""
 			.format(select_fields, conditions, order_by), values, as_dict=True)
 
+		if self.filters.show_future_payments:
+			values.insert(2, self.filters.report_date)
+
+			future_entries = frappe.db.sql("""
```

Comment:
avoid a separate query to get future entries, it takes too much time in old databases.